plasmapy/formulary/parameters.py

```diff
@@ -528,35 +535,54 @@ def ion_sound_speed(
 cs_ = ion_sound_speed
 """ Alias to :func:`ion_sound_speed`. """
 
+# -----                                                             thermal_speed  -----
 
 # This dictionary defines coefficients for thermal speeds
 # calculated for different methods and values of ndim.
 # Created here to avoid re-instantiating on each call
-_coefficients = {
+#
+thermal_speed_coefficients = {
     1: {"most_probable": 0, "rms": 1, "mean_magnitude": 2 / np.pi},
     2: {"most_probable": 1, "rms": 2, "mean_magnitude": np.pi / 2},
     3: {"most_probable": 2, "rms": 3, "mean_magnitude": 8 / np.pi},
 }
+"""
+Dictionary of various coefficients used in calculating the
+`~plasmapy.formulary.parameters.thermal_speed`.
+"""
+
+
+@preserve_signature
+@njit
+def thermal_speed_lite(T, mass, coeff):
+    """
+    A lite weight version of `thermal_speed` intended for computational used
+    and, thus, does not do any argument validation/conditioning.
+    """
+    return np.sqrt(coeff * k_B_si_unitless * T / mass)
 
 
+@add_lite(
+    thermal_speed_lite,
+    attrs=[("coefficients", "thermal_speed_coefficients")],
+    scope=globals(),
+)
 @check_relativistic
 @validate_quantities(
     T={"can_be_negative": False, "equivalencies": u.temperature_energy()},
     mass={"can_be_negative": False, "can_be_nan": True},
 )
 @particles.particle_input
 def thermal_speed(
```

Comment:
The docstring for `thermal_speed` includes a table of the coefficients: maybe we should add a link next to that to the `thermal_speed_coefficients` variable to make more clear that these contain the same information.

Reply:
Agreed! I'll get to this.  I haven't refined and the docstrings because I didn't want to spend the time on it until the framework was approved.